presto-main/src/main/java/com/facebook/presto/metadata/MetadataUtil.java

```diff
@@ -177,7 +178,13 @@ public static Optional<TableHandle> getOptionalTableHandle(Session session, Tran
             ConnectorId connectorId = catalogMetadata.getConnectorId(session, table);
             ConnectorMetadata metadata = catalogMetadata.getMetadataFor(connectorId);
 
-            ConnectorTableHandle tableHandle = metadata.getTableHandle(session.toConnectorSession(connectorId), toSchemaTableName(table));
+            ConnectorTableHandle tableHandle;
+            if (tableVersionExpression.isPresent()) {
+                tableHandle = metadata.getTableHandle(session.toConnectorSession(connectorId), toSchemaTableName(table), tableVersionType, tableVersionExpression);
+            }
+            else {
+                tableHandle = metadata.getTableHandle(session.toConnectorSession(connectorId), toSchemaTableName(table));
+            }
```

Comment:
```suggestion
            tableHandle = tableVersionExpression
                    .map(expression -> metadata.getTableHandle(session.toConnectorSession(connectorId), toSchemaTableName(table), tableVersionType, expression))
                    .orElseGet(() -> metadata.getTableHandle(session.toConnectorSession(connectorId), toSchemaTableName(table)));
```

And then making the type and expression values non-optional